### ISSUE TITLE
Variable gets undefined

### ISSUE THREAD

**Issue by wehmoen** [twigjs/twig.js#751]
Hi,

I'm working on a pretty complex nodeJS using express and twig.js. Been using twig.js for over year now but encountered something strange today. 

In a template, lets call it `watch.twig` i extend `layout.twig`. In `watch.twig` I fill the blocks `meta`, `body` and `script`.  Using express I then run `res.render('watch`, {variable: 123})`

I use `{{ variable }}`  in all three blocks. The variable actually is written in the `meta` and `body` block. But not in the block called `script`.  When I used the `dump()` function in all three block i noticed that the variable is not availbale in this specific block. I upgraded from 0.10.3 to 1.10.4 today and that seems to be the only issue. Before I upgrade twigjs any further I wanted to ask if anyone here ever came across something like this. 

Kind regards

-- The actual templates are much more complex than what I describe here. Just for simplicity:

`layout.twig`

```twig

<div>{% block meta %}{% endblock %}</div>
<div>{% block body %}{% endblock %}</div>
<div>{% block script %}{% endblock %}</div>

```

`watch.twig`

```twig

{% extends './layout.twig' %}

{% block meta %}{{ variable }}{% endblock %}
{% block body %}{{ variable }}{% endblock %}
{% block script %}{{ variable }}{% endblock %}

```

Expected Output:

```
<div>123</div>
<div>123</div>
<div>123</div>
``` 

Actual Output:

```
<div>123</div>
<div>123</div>
<div></div>
```

Simplified express route: 

```js
router.get('/path', (req, res) => {

   res.render('watch', {variable: 123});

});
```

Versions: 

TwigJS: 1.10.4
Express: 4.17.1

**Comment by wehmoen** [comment]
Been able to track it down to one line in the `layout.twig` that uses a macro. A simple macro, that based on the date of the year returned a different string caused these issues. When I removed the macro everything works as expected.  

```twig
{% macro getLogo() %}
  {% set logo_normal = '/img/logo.png' %}
  {% set logo_holiday = '/img/logo_holiday.png' %}
  {% set logo_sylvester = '/img/logo_sylvester.png' %}
  {% set today = ''|date("d") %}
  {% set month = ''|date("m") %}

  {% if month == '12' %}
    {% if today == "24" or today == "25" or today == "26" %}
      {{ logo_holiday }}
    {% elseif today == '31' %}
      {{ logo_sylvester }}
    {% else %}
      {{ logo_normal }}
    {% endif %}
  {% else %}
    {{ logo_normal }}
  {% endif %}

{% endmacro %}
{% import _self as macro %}
```

It was used: `{{macro.getLogo()}}`

Is there anything obviously wrong with it?

**Comment by hctom** [comment]
Same here! I fixed it by moving the macro to another file and importing it via `{% import 'path/to/macros.twig' as macro %}`. From then on, everything works fine again.

**Comment by willrowe** [comment]
Fixed by #773.
